stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -0,0 +1,1375 @@
+                    (block_height,  index_root, consensus_hash,
+                     burn_header_hash, burn_header_height,
+                     burn_header_timestamp, block_size,
+
+                     header_type,
+                     version, chain_length, burn_spent, parent,
+                     burn_view, signature, tx_merkle_root, state_index_root,
+
+                     block_hash,
+                     index_block_hash,
+                     cost,
+                     total_tenure_cost,
+                     tenure_tx_fees,
+                     parent_block_id,
+                     affirmation_weight,
+                     tenure_height,
+                     tenure_changed)
+                    VALUES (?1, ?2, ?3, ?4, ?5, ?6, ?7, ?8, ?9, ?10, ?11, ?12, ?13, ?14, ?15, ?16, ?17, ?18, ?19, ?20, ?21, ?22, ?23, ?24, ?25)",
+            args
+        )?;
+
+        Ok(())
+    }
+
+    /// Append a Stacks block to an existing Stacks block, and grant the miner the block reward.
+    /// Return the new Stacks header info.
+    pub fn advance_tip(
+        headers_tx: &mut StacksDBTx,
+        parent_tip: &StacksBlockHeaderTypes,
+        parent_consensus_hash: &ConsensusHash,
+        new_tip: &NakamotoBlockHeader,
+        new_consensus_hash: &ConsensusHash,
+        new_burn_header_hash: &BurnchainHeaderHash,
+        new_burnchain_height: u32,
+        new_burnchain_timestamp: u64,
+        block_reward: Option<&MinerPaymentSchedule>,
+        mature_miner_payouts: Option<(MinerReward, Vec<MinerReward>, MinerReward, MinerRewardInfo)>, // (miner, [users], parent, matured rewards)
+        anchor_block_cost: &ExecutionCost,
+        total_tenure_cost: &ExecutionCost,
+        anchor_block_size: u64,
+        applied_epoch_transition: bool,
+        burn_stack_stx_ops: Vec<StackStxOp>,
+        burn_transfer_stx_ops: Vec<TransferStxOp>,
+        burn_delegate_stx_ops: Vec<DelegateStxOp>,
+        affirmation_weight: u64,
+        tenure_height: u64,
+        tenure_changed: bool,
+        block_fees: u128,
+    ) -> Result<StacksHeaderInfo, ChainstateError> {
+        if new_tip.parent != FIRST_STACKS_BLOCK_HASH {
+            // not the first-ever block, so linkage must occur
+            assert_eq!(new_tip.parent, parent_tip.block_hash());
+        }
+
+        assert_eq!(
+            parent_tip
+                .height()
+                .checked_add(1)
+                .expect("Block height overflow"),
+            new_tip.chain_length
+        );
+
+        let parent_hash =
+            StacksChainState::get_index_hash(parent_consensus_hash, &parent_tip.block_hash());
+
+        let new_block_hash = new_tip.block_hash();
+        let index_block_hash = StacksBlockId::new(&new_consensus_hash, &new_block_hash);
+
+        // store each indexed field
+        test_debug!(
+            "Headers index_put_begin {}-{}",
+            &parent_hash,
+            &index_block_hash,
+        );
```

Comment:
I think we should prefer inline variables in format strings, it's easier to read:

```suggestion
        test_debug!(
            "Headers index_put_begin {parent_hash}-{index_block_hash}"
        );
```